src/main/kotlin/no/nav/syfo/service/VarselSendtService.kt

```diff
@@ -0,0 +1,48 @@
+package no.nav.syfo.service
+
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.consumer.PdlConsumer
+import no.nav.syfo.consumer.SyfosyketilfelleConsumer
+import no.nav.syfo.db.DatabaseInterface
+import no.nav.syfo.db.domain.PPlanlagtVarsel
+import no.nav.syfo.db.domain.VarselType
+import no.nav.syfo.db.fetchUtsendtVarselByFnr
+import no.nav.syfo.utils.dateIsInInterval
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.lang.RuntimeException
+import java.time.LocalDate
+
+class VarselSendtService(
+    val pdlConsumer: PdlConsumer,
+    val syfosyketilfelleConsumer: SyfosyketilfelleConsumer,
+    val databaseAccess: DatabaseInterface
+) {
+    private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.db.VarselSendtService")
+
+    fun erVarselSendt(aktorId: String, varselType: VarselType) : Boolean {
+        val fnr = pdlConsumer.getFnr(aktorId) ?: return false
+        val syketilfelle = runBlocking {
+            syfosyketilfelleConsumer.getOppfolgingstilfelle39Uker(aktorId)
+        }
+        return syketilfelle?.let {
+            val fom = syketilfelle.fom
+            val tom = syketilfelle.tom
+            return erVarselSendt(fnr, varselType, fom, tom)
+        } ?: false
+    }
+
+    fun erVarselSendt(fnr: String, varselType: VarselType, tilfelleFom: LocalDate, tilfelleTom: LocalDate): Boolean {
+        val utsendtVarsel = databaseAccess.fetchUtsendtVarselByFnr(fnr)
+            .filter { it.type == varselType.name }
+
+        if (utsendtVarsel.isEmpty()) {
+            return false
+        }
+
+        utsendtVarsel.sortedBy { it.utsendtTidspunkt }
```

Comment:
utsendtVarsel.sortedBy { it.utsendtTidspunkt } muterer vel ikke opprinnelige data, så denne gjør kanskje ikke noe?

Reply:
..eventuelt ha en val sortedData = utsendtVarsel.sortedBy { it.utsendtTidspunkt }

Reply:
👍